meson.build

```diff
@@ -0,0 +1,1629 @@
+
+shaderc = {
+    'name': 'shaderc',
+    'use': shaderc_shared['use'] or shaderc_static['use']
+}
+if shaderc['use']
+    sources += 'video/out/gpu/spirv_shaderc.c'
+endif
+
+sixel = dependency('sixel', version: '>= 1.5', required: get_option('sixel'))
+if sixel.found()
+    dependencies += sixel
+    sources += 'video/out/vo_sixel.c'
+endif
+
+spirv_cross_shared = {
+    'name': 'spirv-cross-shared',
+    'deps': dependency('spirv-cross-c-shared', required: get_option('spirv-cross')),
+}
+spirv_cross_shared += {'use': spirv_cross_shared['deps'].found() and get_option('default_library') == 'shared'}
+
+spirv_cross_static = {
+    'name': 'spirv-cross-static',
+    'deps': dependency('spirv-cross', required: get_option('spirv-cross')),
+}
+spirv_cross_static += {'use': spirv_cross_static['deps'].found() and not spirv_cross_shared['use']}
+
+spirv_cross = {
+    'name': 'spirv-cross',
+    'use': spirv_cross_shared['use'] or spirv_cross_static['use']
+}
+
+d3d11 = {
+    'name': 'd3d11',
+    'use': win32_desktop and shaderc['use'] and spirv_cross['use'] and not get_option('d3d11').disabled(),
+}
+
+wayland = {
+    'name': 'wayland',
+    'deps': [dependency('wayland-client', version: '>= 1.15.0', required: get_option('wayland')),
+             dependency('wayland-cursor', version: '>= 1.15.0', required: get_option('wayland')),
+             dependency('wayland-protocols', required: get_option('wayland')),
+             dependency('xkbcommon', version: '>= 0.3.0', required: get_option('wayland'))],
+    'header': cc.check_header('linux/input-event-codes.h'),
+    'scanner': find_program('wayland-scanner', required: get_option('wayland')),
+    'use': false,
+}
+wayland_deps = true
+foreach dep: wayland['deps']
+    if not dep.found()
+        wayland_deps = false
+        break
+    endif
+endforeach
+
+if wayland_deps and wayland['header'] and wayland['scanner'].found()
+    wayland += {'use': true}
+endif
+
+if wayland['use']
+    wl_protocol_dir = wayland['deps'][2].get_pkgconfig_variable('pkgdatadir')
+    protocols = [[wl_protocol_dir, 'stable/xdg-shell/xdg-shell.xml'],
+                 [wl_protocol_dir, 'stable/presentation-time/presentation-time.xml'],
+                 [wl_protocol_dir, 'unstable/idle-inhibit/idle-inhibit-unstable-v1.xml'],
+                 [wl_protocol_dir, 'unstable/xdg-decoration/xdg-decoration-unstable-v1.xml']]
+    wl_protocols_source = []
+    wl_protocols_headers = []
+
+    wayland_dir = join_paths(generated_dir, 'wayland')
+    run_command('mkdir', '-p', wayland_dir)
+    foreach p: protocols
+        xml = join_paths(p)
+        protocol_name = run_command('basename', xml, '.xml')
+        protocol_code = join_paths(wayland_dir, protocol_name.stdout().strip() + '.c')
+        protocol_header = join_paths(wayland_dir, protocol_name.stdout().strip() + '.h')
+        run_command('wayland-scanner', 'private-code', xml, protocol_code)
+        run_command('wayland-scanner', 'client-header', xml, protocol_header)
```

Comment:
custom_target is superior to run_command, again, because if wayland-protocols gets updated then a custom target will regenerate fresh copies of the files. run_command, again, only runs during initial project setup.